internal/core/operations/healthcontroller/health_controller_executor.go

```diff
@@ -0,0 +1,250 @@
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package healthcontroller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/topfreegames/maestro/internal/core/services/room_manager"
+
+	"github.com/topfreegames/maestro/internal/core/entities"
+	"github.com/topfreegames/maestro/internal/core/entities/game_room"
+	"github.com/topfreegames/maestro/internal/core/logs"
+	"github.com/topfreegames/maestro/internal/core/operations/add_rooms"
+	"github.com/topfreegames/maestro/internal/core/operations/remove_rooms"
+
+	"github.com/topfreegames/maestro/internal/core/ports"
+
+	"github.com/topfreegames/maestro/internal/core/entities/operation"
+	"github.com/topfreegames/maestro/internal/core/operations"
+	"go.uber.org/zap"
+)
+
+type SchedulerHealthControllerExecutor struct {
+	roomStorage       ports.RoomStorage
+	instanceStorage   ports.GameRoomInstanceStorage
+	schedulerStorage  ports.SchedulerStorage
+	operationManager  ports.OperationManager
+	roomManagerConfig room_manager.RoomManagerConfig
+}
+
+var _ operations.Executor = (*SchedulerHealthControllerExecutor)(nil)
+
+func NewExecutor(roomStorage ports.RoomStorage, instanceStorage ports.GameRoomInstanceStorage, schedulerStorage ports.SchedulerStorage, operationManager ports.OperationManager, roomManagerConfig room_manager.RoomManagerConfig) *SchedulerHealthControllerExecutor {
+	return &SchedulerHealthControllerExecutor{
+		roomStorage:       roomStorage,
+		instanceStorage:   instanceStorage,
+		schedulerStorage:  schedulerStorage,
+		operationManager:  operationManager,
+		roomManagerConfig: roomManagerConfig,
+	}
+}
+
+func (ex *SchedulerHealthControllerExecutor) Execute(ctx context.Context, op *operation.Operation, definition operations.Definition) operations.ExecutionError {
+	logger := zap.L().With(
+		zap.String(logs.LogFieldSchedulerName, op.SchedulerName),
+		zap.String(logs.LogFieldOperationDefinition, op.DefinitionName),
+		zap.String("operation_phase", "Execute"),
+		zap.String(logs.LogFieldOperationID, op.ID),
+	)
+
+	gameRoomIDs, instances, scheduler, err := ex.loadActualState(ctx, op, logger)
+	if err != nil {
+		return operations.NewErrUnexpected(err)
+	}
+
+	nonexistentGameRoomIDs := ex.checkNonexistentGameRoomsIDs(gameRoomIDs, instances)
+	if len(nonexistentGameRoomIDs) > 0 {
+		logger.Error("found registered rooms that no longer exists")
+		ex.tryEnsureCorrectRoomsOnStorage(ctx, op, logger, nonexistentGameRoomIDs)
+	}
+
+	existentGameRoomIDs := difference(gameRoomIDs, nonexistentGameRoomIDs)
```

Comment:
What do you think to rename this function to something more semantic like `filterExistentGameRooms`